src/testutils/blockchain_tests.rs

```diff
@@ -1059,20 +1072,25 @@ macro_rules! bdk_blockchain_tests {
                 let wallet_addr = wallet.get_address($crate::wallet::AddressIndex::New).unwrap().address;
 
                 wallet.sync(&blockchain, SyncOptions::default()).unwrap();
-                assert_eq!(wallet.get_balance().unwrap(), 0, "incorrect balance");
+                assert_eq!(wallet.get_balance().unwrap().immature, 0, "incorrect balance");
 
                 test_client.generate(1, Some(wallet_addr));
 
-                #[cfg(feature = "rpc")]
+                wallet.sync(&blockchain, SyncOptions::default()).unwrap();
+
+                #[cfg(not(feature="rpc"))]
```

Comment:
I swear I was about to merge when I noticed this...

Why is this line here? Shouldn't we just have an assert of the immature balance for every backend (not only "not rpc"), and then the code for making it mature?

```
// we consider the coinbase only when mature (100 blocks)
assert!(wallet.get_balance().unwrap().immature > 0, "incorrect balance after receiving coinbase");

// make it mature
let node_addr = test_client.get_node_address(None);
test_client.generate(100, Some(node_addr));
```

Reply:
That is so because, what I could get from the previous code (and the accompanying comment) is that rpc doesn't report immature coins in balance. A quick test without it shows exactly that. Before the code only "matured" coinbase if backend in use was rpc, now I just test immature coins in all cases other that rpc, and mature coins anyway.

Reply:
Ok, but at this point if you drop the `#[cfg(not(feature="rpc"))]` and check the immature for the rpc as well, the tests pass, don't they?

Reply:
Nope. Balance is reported as 0, 0, 0, 0.

Reply:
Ouch, yeah, now I see it. You won't be happy to hear this, but I really think this should be fixed before merging...